circuit_maintenance_parser/parsers/att.py

```diff
@@ -0,0 +1,114 @@
+    r"Business Risk: (.*)"
+)
+RE_MAINTENANCE_WINDOW_GMT = re.compile(r"Start Time: (.* GMT).*End Time: (.* GMT)")
+RE_MAINTENANCE_WINDOW_NO_TIMEZONE = re.compile(r"Start Time: (.*)[ \n]End Time: (.*)")
+
+
+class XlsxParserATT1(Xlsx):
+    """Xlsx Parser for ATT file attachments."""
+
+    @staticmethod
+    def parse_xlsx(records: List[Dict]) -> List[Dict]:
+        """Parses ATT xlsx attachments."""
+        impact = Impact.OUTAGE
+        account_name, circuit_id_key = get_account_and_circuit_id_key(records[0])
+        circuit_ids = [r[circuit_id_key] for r in records]
+        if "Customer" in records[0]:
+            circuit_ids = [normalize_lec_circuit_id(cid) for cid in circuit_ids]
+        circuits = [CircuitImpact(impact=impact, circuit_id=cid) for cid in circuit_ids]
+        data = [
+            {
+                "account": account_name,
+                "circuits": circuits,
+            }
+        ]
+        return data
+
+
+class HtmlParserATT1(Html):
+    """Notifications Parser for ATT notifications."""
+
+    def parse_html(self, soup):
+        """Parse ATT HTML notification."""
+        logger.debug("Parsing ATT HTML notification.")
+        data = self.parse_p_tags(soup)
+        data["start"] = self.dt2ts(data["start"])
+        data["end"] = self.dt2ts(data["end"])
+        data["status"] = Status.CONFIRMED
+        return [data]
+
+    @staticmethod
+    def parse_p_tags(soup: ResultSet) -> Dict:
+        """Parse <p> tags in HTML."""
+        data = {}
+        p_tags = soup.find_all("p")
+
+        for tag in p_tags:
+            text = tag.text.strip()
+            text = remove_unprintable(text)
+
+            if match := RE_EVENT.search(text):
+                event_id, impact, summary, description, _ = match.groups()
+                data["maintenance_id"] = event_id
+                data["summary"] = f"{summary}: {impact} {description}"
+
+            elif match := RE_MAINTENANCE_WINDOW_GMT.search(text):
+                start_time_text, end_time_text = match.groups()
+                data["start"] = dateutil.parser.parse(start_time_text)
+                data["end"] = dateutil.parser.parse(end_time_text)
+
+            elif match := RE_MAINTENANCE_WINDOW_NO_TIMEZONE.search(text):
+                start_time_text, end_time_text = match.groups()
+                data["start"] = dateutil.parser.parse(start_time_text + " GMT")
+                data["end"] = dateutil.parser.parse(end_time_text + " GMT")
+
+        return data
+
+
+def get_account_and_circuit_id_key(record: Dict) -> tuple[str, str]:
+    """Return the account name and the key used to retrieve circuits IDs.
+
+    The key names may vary depending on the ATT business unit that initiated the notice.
+    """
+    if account := record.get("Customer"):
+        circuit_id_key = "Circuit/Asset"
+    elif account := record.get("Customer Name"):
+        circuit_id_key = "Circuit ID"
+    elif account := record.get("Customer Names"):
```

Comment:
does it mean that there could be different accounts in the same maintenance?

Reply:
notice that you can detect multiple maintenances per notification

Reply:
I have only ever seen a single account with multiple circuits specified.